ioctl/newcmd/account/accountdelete.go

```diff
@@ -99,43 +98,53 @@ func NewAccountDelete(c ioctl.Client) *cobra.Command {
 			if err != nil {
 				return output.NewError(output.ConvertError, failToConvertStringIntoAddress, nil)
 			}
-			ks := c.NewKeyStore(config.ReadConfig.Wallet, keystore.StandardScryptN, keystore.StandardScryptP)
-			for _, v := range ks.Accounts() {
-				if bytes.Equal(account.Bytes(), v.Address.Bytes()) {
-					var confirm string
-					message := output.ConfirmationMessage{Info: infoWarn, Options: []string{"yes"}}
-					fmt.Println(message.String())
-					fmt.Scanf("%s", &confirm)
-					if !strings.EqualFold(confirm, "yes") {
-						output.PrintResult("quit")
-						return nil
-					}
 
-					if err := os.Remove(v.URL.Path); err != nil {
-						return output.NewError(output.WriteFileError, failToRemoveKeystoreFile, err)
+			var filePath string
+			if CryptoSm2 {
+				if filePath == "" {
+					filePath = filepath.Join(c.Config().Wallet, "sm2sk-"+account.String()+".pem")
+				}
+			} else {
+				ks := c.NewKeyStore(c.Config().Wallet, keystore.StandardScryptN, keystore.StandardScryptP)
+				for _, v := range ks.Accounts() {
+					if bytes.Equal(account.Bytes(), v.Address.Bytes()) {
+						filePath = v.URL.Path
+						break
 					}
+				}
+			}
 
-					aliases := make(map[string]string)
-					for name, addr := range c.Config().Aliases {
-						aliases[addr] = name
-					}
+			// check whether crypto file exists
+			if _, err = os.Stat(filePath); err != nil {
+				return output.NewError(output.ValidationError, fmt.Sprintf(failToFindAccount, addr), nil)
+			}
+			var confirm string
+			message := output.ConfirmationMessage{Info: infoWarn, Options: []string{"yes"}}
+			fmt.Println(message.String())
+			fmt.Scanf("%s", &confirm)
+			if !strings.EqualFold(confirm, "yes") {
+				output.PrintResult("quit")
+				return nil
+			}
 
-					delete(config.ReadConfig.Aliases, aliases[addr])
-					out, err := yaml.Marshal(&config.ReadConfig)
-					if err != nil {
-						return output.NewError(output.SerializationError, "", err)
-					}
-					if err := ioutil.WriteFile(config.DefaultConfigFile, out, 0600); err != nil {
-						return output.NewError(output.WriteFileError,
-							fmt.Sprintf(failToWriteToConfigFile, config.DefaultConfigFile), err)
-					}
-					output.PrintResult(fmt.Sprintf(resultSuccess, addr))
-					return nil
-				}
+			if err := os.Remove(filePath); err != nil {
+				return output.NewError(output.WriteFileError, failToRemoveKeystoreFile, err)
 			}
-			return output.NewError(output.ValidationError, fmt.Sprintf(failToFindAccount, addr), nil)
+
+			aliases := c.GetAliasMap()
+			cfg := c.Config()
+			delete(cfg.Aliases, aliases[addr])
+			out, err := yaml.Marshal(&cfg)
+			if err != nil {
+				return output.NewError(output.SerializationError, "", err)
+			}
+			if err := ioutil.WriteFile(config.DefaultConfigFile, out, 0600); err != nil {
```

Comment:
As of Go 1.16, this function  simply calls `io.WriteFile`, direct use `io` package instead of `ioutil`.

Reply:
good!